src/Worker.ts

```diff
@@ -0,0 +1,37 @@
+import { ref, onMounted, onUnmounted, Ref } from '@vue/composition-api';
+
+export function useWorker(url: string) {
+  let message: Ref<any> = ref(null); 
+  let ww: Worker;
+  
+  const post: typeof ww.postMessage = function post(val: any) {
+    if (!ww) return;
+
+    ww.postMessage(val);
+  };
+
+  const terminate: typeof ww.terminate = function terminate() {
+    if (!ww) return;
+
+    ww.terminate();
+  };
+
+  onMounted(() => {
+    ww = new Worker(url);
+
+    ww.onmessage = (e: MessageEvent) => {
+      message.value = e.data;
+      ww.postMessage(message);
```

Comment:
I don't think this is correct. The `message` would be received from the web worker, why are we sending it back?

Reply:
This is what I am trying to figure out. 
I went to [this Mozilla Guide](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) to figure out how the Worker should be implemented. I believe that **"Sending messages to and from dedicated worker"** is what to be implemented here. 

But I think I got it after your clarification in the Issue Comments.